client.go

```diff
@@ -1375,3 +1397,545 @@ func (c *ToznySDKV3) UnbrokerShare(ctx context.Context, authorizerClientID strin
+		EncryptedFileName: encryptedFileName,
+		Checksum:          checksum,
+		Size:              size,
+	}
+	uploadResp, err := file.UploadFile(uploadRequest)
+	if err != nil || uploadResp != 0 {
+		return nil, err
+	}
+	// Register the file as being written
+	response, err := c.StorageClient.FileCommit(ctx, pendingFileURL.PendingFileID)
+	if err != nil {
+		return nil, err
+	}
+	// get the file from the record
+	fileSecret := &pdsClient.Record{
+		Metadata: pdsClient.Meta{
+			RecordID:     response.Metadata.RecordID.String(),
+			WriterID:     response.Metadata.WriterID.String(),
+			UserID:       response.Metadata.UserID.String(),
+			Type:         response.Metadata.Type,
+			Plain:        response.Metadata.Plain,
+			Created:      response.Metadata.Created,
+			LastModified: response.Metadata.LastModified,
+			Version:      response.Metadata.Version.String(),
+			FileMeta:     (*pdsClient.FileMeta)(response.Metadata.FileMeta),
+		},
+		Data:            response.Data,
+		RecordSignature: response.RecordSignature,
+	}
+	return fileSecret, nil
+}
+
+// WriteRecord encrypts the data for the record and creates a new record in E3DB
+func (c *ToznySDKV3) WriteRecord(ctx context.Context, data map[string]string, recordType string, plain map[string]string) (*pdsClient.Record, error) {
+	recordToWrite := pdsClient.WriteRecordRequest{
+		Data: data,
+		Metadata: pdsClient.Meta{
+			Type:     recordType,
+			WriterID: c.E3dbPDSClient.ClientID,
+			UserID:   c.E3dbPDSClient.ClientID,
+			Plain:    plain,
+		},
+	}
+	encryptedRecord, err := c.E3dbPDSClient.EncryptRecord(ctx, recordToWrite)
+	if err != nil {
+		return nil, err
+	}
+	recordSecret, err := c.E3dbPDSClient.WriteRecord(ctx, encryptedRecord)
+	if err != nil {
+		return nil, err
+	}
+	return recordSecret, nil
+}
+
+func (c *ToznySDKV3) ShareRecordWithGroup(ctx context.Context, recordType string, group *storageClient.Group) error {
+	keyReq := pdsClient.GetOrCreateAccessKeyRequest{
+		WriterID:   c.E3dbPDSClient.ClientID,
+		UserID:     c.E3dbPDSClient.ClientID,
+		ReaderID:   c.E3dbPDSClient.ClientID,
+		RecordType: recordType,
+	}
+	keyResp, err := c.E3dbPDSClient.GetOrCreateAccessKey(ctx, keyReq)
+	if err != nil {
+		return err
+	}
+	encryptionKeys := e3dbClients.EncryptionKeys{
+		Public: e3dbClients.Key{
+			Type:     e3dbClients.DefaultEncryptionKeyType,
+			Material: c.StorageClient.EncryptionKeys.Public.Material,
+		},
+		Private: c.StorageClient.EncryptionKeys.Private,
+	}
+	wrappedAK, err := e3dbClients.EncryptAccessKey(keyResp, encryptionKeys)
+	if err != nil {
+		return err
+	}
+	accessKeyReq := storageClient.GroupAccessKeyRequest{
+		GroupID:            group.GroupID,
+		RecordType:         recordType,
+		EncryptedAccessKey: wrappedAK,
+		PublicKey:          group.PublicKey,
+	}
+	_, encryptedGroupAK, err := c.StorageClient.CreateGroupAccessKey(ctx, accessKeyReq)
+	if err != nil {
+		return err
+	}
+	// share secret with group
+	secretShareReq := storageClient.ShareGroupRecordRequest{
```

Comment:
nit: I won't let you save more time by leaving off `uest` because now you have to read this comment. But you can always ignore me, this is a nit and not a blocker.